src/Compiler/CodeGen/EraseUnions.fs

```diff
@@ -19,196 +19,214 @@ let TagCons = 1
+            then
                 RuntimeTypes
             else
                 IntegerTag
 
-    // WARNING: this must match IsUnionTypeWithNullAsTrueValue in the F# compiler 
-    member repr.RepresentAlternativeAsNull (cu,alt) = 
+    // WARNING: this must match IsUnionTypeWithNullAsTrueValue in the F# compiler
+    member repr.RepresentAlternativeAsNull(cu, alt) =
         let alts = getAlternatives cu
-        nullPermitted cu &&
-        (repr.DiscriminationTechnique cu  = RuntimeTypes) && (* don't use null for tags, lists or single-case  *)
-        Array.existsOne isNullary alts  &&
-        Array.exists (isNullary >> not) alts  &&
-        isNullary alt  (* is this the one? *)
 
-    member repr.RepresentOneAlternativeAsNull cu = 
+        nullPermitted cu
+        && (repr.DiscriminationTechnique cu = RuntimeTypes)
+        && (* don't use null for tags, lists or single-case  *) Array.existsOne isNullary alts
+        && Array.exists (isNullary >> not) alts
+        && isNullary alt (* is this the one? *)
+
+    member repr.RepresentOneAlternativeAsNull cu =
         let alts = getAlternatives cu
-        nullPermitted cu &&
-        alts |> Array.existsOne (fun alt -> repr.RepresentAlternativeAsNull (cu,alt))
 
-    member repr.RepresentSingleNonNullaryAlternativeAsInstancesOfRootClassAndAnyOtherAlternativesAsNull (cu,alt) = 
-        // Check all nullary constructors are being represented without using sub-classes 
+        nullPermitted cu
+        && alts |> Array.existsOne (fun alt -> repr.RepresentAlternativeAsNull(cu, alt))
+
+    member repr.RepresentSingleNonNullaryAlternativeAsInstancesOfRootClassAndAnyOtherAlternativesAsNull(cu, alt) =
+        // Check all nullary constructors are being represented without using sub-classes
         let alts = getAlternatives cu
-        not (isStruct cu) &&
-        not (isNullary alt) &&
-        (alts |> Array.forall (fun alt2 -> not (isNullary alt2) || repr.RepresentAlternativeAsNull (cu,alt2))) &&
-        // Check this is the one and only non-nullary constructor 
+
+        not (isStruct cu)
+        && not (isNullary alt)
+        && (alts
+            |> Array.forall (fun alt2 -> not (isNullary alt2) || repr.RepresentAlternativeAsNull(cu, alt2)))
+        &&
+        // Check this is the one and only non-nullary constructor
         Array.existsOne (isNullary >> not) alts
 
-    member repr.RepresentAlternativeAsStructValue cu = 
-        isStruct cu 
+    member repr.RepresentAlternativeAsStructValue cu = isStruct cu
 
-    member repr.RepresentAlternativeAsFreshInstancesOfRootClass (cu,alt) = 
-       not (isStruct cu) && 
-       (// Check all nullary constructors are being represented without using sub-classes 
-        (isList cu  && nameOfAlt alt = ALT_NAME_CONS) ||
-        repr.RepresentSingleNonNullaryAlternativeAsInstancesOfRootClassAndAnyOtherAlternativesAsNull (cu, alt) )
+    member repr.RepresentAlternativeAsFreshInstancesOfRootClass(cu, alt) =
```

Comment:
Wow!!! below is truly ugly ....  why on earth did it format like this?
Oic the inline comment was moved down a row and following the func, dividing the func from its arguments.